README.md

```diff
@@ -1,67 +1,76 @@
-# SetReplace
+# Set Substitution System
 
-## Set | Network Substitution Systems
+## Basic Example
 
-**SetReplace** is a Wolfram Language package that implements a substitution system such that in each step an unordered subset matching a given pattern is deleted from a multiset and replaced with different subset. If each element of the set consists of pairs of elements, this set can be thought of as a directed graph, and the system becomes a ***network substitution*** (aka *graph rewrite*) ***system***.
-
-An example of such a system would be removing vertices from a path graph
+**SetReplace** is a [Wolfram Language](https://www.wolfram.com/language/) package to manipulate set substitution systems. To understand what a set substitution system does consider an unordered set of elements:
 ```
-In[.] := SetReplace[{{0, 1}, {1, 2}, {2, 3}}, {{a_, b_}, {b_, c_}} :> {{a, c}}, 2]
+{1, 2, 5, 3, 6}
 ```
+We can set up an operation on this set which would take any of the two elements and replace them with their sum:
 ```
-Out[.] = {{0, 3}}
+{a_, b_} :> {a + b}
 ```
-
-Here, `{{0, 1}, {1, 2}, {2, 3}}` is the initial condition, `{{a_, b_}, {b_, c_}} :> {{a, c}}` is the rule that matches pairs of edges sharing a vertex and replaces them with a single edge, and `2` the number of steps, i.e., the number of times the rule is applied.
-
-One can also imagine an inverse of the rule above
+In **SetReplace**, this can be expressed as (the new element is put at the end)
 ```
-In[.] := SetReplace[{{0, 1}}, {{a_, b_}} :>
-  Module[{new}, {{a, new}, {new, b}}], 2]
+In[] := SetReplace[{1, 2, 5, 3, 6}, {a_, b_} :> {a + b}]
+Out[] = {5, 3, 6, 3}
 ```
+Note that this is similar to `SubsetReplace` function of Wolfram Language (which did not exist prior to version 12.1, and which by default replaces all non-overlapping subsets at once)
 ```
-Out[.] = {{v11, 1}, {0, v12}, {v12, v11}}
+In[] := SubsetReplace[{1, 2, 5, 3, 6}, {a_, b_} :> Sequence[a + b]]
+Out[] = {3, 8, 6}
 ```
 
-We use `Module` here to assign unique names to the newly created vertices. As using `Module` is somewhat cumbersome, a simplified syntax is available, where integers are automatically converted into patterns, and `Module` is added if necessary
+## Relations between Set Elements
+
+A more interesting case (and the only case we have studied with any reasonable detail) is the case of set elements that are related to each other. Specifically, the elements can be expressed as ordered lists of atoms (or vertices), and the set essentially becomes an ordered hypergraph.
+
+As a simple example consider a set
 ```
-In[.] := SetReplace[{{0, 1}},
- Echo @ FromAnonymousRules[{{0, 1}} -> {{0, 2}, {2, 1}}], 2]
+{{1, 2, 3}, {2, 4, 5}, {4, 6, 7}}
 ```
+which we can render as a collection of ordered hyperedges:
 ```
->> {{v1_, v2_}} :> Module[{v3}\, {{v1, v3}, {v3, v2}}]
-Out[.] = {{v13, 1}, {0, v14}, {v14, v13}}
+In[] := HypergraphPlot[{{1, 2, 3}, {2, 4, 5}, {4, 6, 7}},
```

Comment:
Using HypergraphPlot on purpose?

Reply:
Yes, because I think `WolframModelPlot` would be confusing given that I have not introduced not only `WolframModel`, but even any kind of evolution at this point. Do you think it's a bad idea?

Reply:
Can you put the image there without showing the input. And you can say you will talk about plotting them in detail later

Reply:
I would prefer not to. For consistency, I want to keep the inputs for all images. I think it would be better to just start using `WolframModelPlot` right away. Ideally, I would mention that `WolframModelPlot` supports plotting other features of `WolframModel`s, which is why it's called this way, but of course it does not do that yet. Or, as a crazy idea, I can not use `WolframModelPlot` at all, and use `HypergraphPlot` everywhere, and adjust its usage message to not say it's deprecated.

Reply:
I think this is good as is.. I am going to resolve this